### ISSUE TITLE
Upgrading to sdk version 8.3.2 causes auth errors against firestore emulator

### ISSUE THREAD

**Issue by zwily** [firebase/firebase-js-sdk#4721]
### [REQUIRED] Describe your environment

  * Operating System version: OS X 11.2.3
  * Browser version: Any
  * Firebase SDK version: 8.3.2
  * Firebase Product: auth/firestore (?)

#### Steps to reproduce:

I upgraded from 8.2.5 to 8.3.2, and after upgrading, found that any calls to my firestore emulator were failing with auth failures. Authentication (Google auth against the auth emulator) does appear to succeed. 

I found the following blob in `firestore-debug.log` that does not appear when using 8.3.1. (Token is just against emulator, not sensitive.)

```
Apr 03, 2021 3:04:32 PM com.google.net.webchannel.server.common.CorsFilter populateCustomHeaders
WARNING: Invalid $httpHeaders: X-Goog-Api-Client:gl-js/ fire/8.3.2
X-Firebase-GMPID:1:421228592265:web:2b314c803b3eccc93d1a58
Content-Type:text/plain
Authorization:Bearer eyJhbGciOiJub25lIiwidHlwIjoiSldUIn0.eyJuYW1lIjoiSm9lIEJvYiIsInJvbGVzIjpbInVzZXIiLCJhZG1pbiIsImF1dGhvciIsIm1lbWVhcHByb3ZlciIsImFwcHJvdmVyIiwiZWRpdG9yIl0sImh0dHBzOi8vaGFzdXJhLmlvL2p3dC9jbGFpbXMiOnsieC1oYXN1cmEtZGVmYXVsdC1yb2xlIjoidXNlciIsIngtaGFzdXJhLWFsbG93ZWQtcm9sZXMiOlsidXNlciIsImFkbWluIiwiYXV0aG9yIiwibWVtZWFwcHJvdmVyIiwiYXBwcm92ZXIiLCJlZGl0b3IiXSwieC1oYXN1cmEtdXNlci1pZCI6IkIyU0FLdVNZeTRjMWFEbWxKU3ZzQ1BjRkQxejgifSwiZW1haWwiOiJqb2VAZXhhbXBsZS5jb20iLCJlbWFpbF92ZXJpZmllZCI6dHJ1ZSwiYXV0aF90aW1lIjoxNjE3NDgzODcyLCJ1c2VyX2lkIjoiQjJTQUt1U1l5NGMxYURtbEpTdnNDUGNGRDF6OCIsImZpcmViYXNlIjp7ImlkZW50aXRpZXMiOnsiZW1haWwiOlsiam9lQGV4YW1wbGUuY29tIl0sImdvb2dsZS5jb20iOlsiMDc2Mjg2ODc0MDM4NTQ4ODAzMzI1MDk0OTE5NTE1NjIyODUyODE3OSJdfSwic2lnbl9pbl9wcm92aWRlciI6Imdvb2dsZS5jb20ifSwiaWF0IjoxNjE3NDgzODcyLCJleHAiOjE2MTc0ODc0NzIsImF1ZCI6ImJoci1jbXMiLCJpc3MiOiJodHRwczovL3NlY3VyZXRva2VuLmdvb2dsZS5jb20vYmhyLWNtcyIsInN1YiI6IkIyU0FLdVNZeTRjMWFEbWxKU3ZzQ1BjRkQxejgifQ.
```

I bisected all the intermediate firebase-sdk upgrades, and it was the move from 8.3.1 -> 8.3.2 that started doing this. I am also using the latest `firebase-tools` (9.8.0).

I scanned through the diff of 8.3.1 -> 8.3.2 but the refactoring looked fairly extensive.

#### Relevant Code:

Any authenticated firestore read against emulator. I'm trying to put together a bare app demonstrating the problem and will update the ticket if I manage to get that.

**Comment by zwily** [comment]
I created a small app demonstrating the issue:

https://github.com/zwily/firebase-8.3.2-test

Somehow, auth information isn't making it into the firestore emulator. The same problem does not happen against production. This may be an emulator bug, but it does not happen with the current emulators and 8.3.1.

**Comment by tomhicks** [comment]
I found this issue as I have been having issues with permissions. My previously-working Firestore rules are now failing with a "Null value error" saying I cannot create a document where I could before.

Downgrading to 8.3.1 fixes the issue.

**Comment by yuchenshi** [comment]
See https://github.com/firebase/firebase-tools/issues/3258#issuecomment-814402977

**Comment by yuchenshi** [comment]
This should be fixed in Firebase CLI v9.9.0.
